components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -549,21 +551,112 @@ public static String getAccessTokenPartitioningDomains() {
         return userStoreDomainMap;
     }
 
-    public static String getUserStoreDomainFromUserId(String userId)
+    public static String getMappedUserStoreDomain(String userStoreDomain) throws IdentityOAuth2Exception {
+
+        String mappedUserStoreDomain = userStoreDomain;
+
+        Map<String, String> availableDomainMappings = OAuth2Util.getAvailableUserStoreDomainMappings();
+        if (availableDomainMappings.containsKey(userStoreDomain)) {
+            mappedUserStoreDomain = availableDomainMappings.get(userStoreDomain);
+        }
+
+        return mappedUserStoreDomain;
+    }
+
+    public static String getPartitionedTableByUserStore(String tableName, String userStoreDomain)
             throws IdentityOAuth2Exception {
+
+        if (StringUtils.isNotBlank(userStoreDomain) &&
+                !IdentityUtil.getPrimaryDomainName().equalsIgnoreCase(userStoreDomain)) {
+
+            String mappedUserStoreDomain = OAuth2Util.getMappedUserStoreDomain(userStoreDomain);
+            tableName = tableName + "_" + mappedUserStoreDomain;
+        }
+
+        return tableName;
+    }
+
+    public static String getTokenPartitionedSqlByUserStore(String sql, String userStoreDomain)
+            throws IdentityOAuth2Exception {
+
+        String partitionedSql = sql;
+
+        if (OAuth2Util.checkAccessTokenPartitioningEnabled() && OAuth2Util.checkUserNameAssertionEnabled()) {
+
+            String partitionedAccessTokenTable = OAuth2Util.getPartitionedTableByUserStore(OAuthConstants.
+                    ACCESS_TOKEN_STORE_TABLE, userStoreDomain);
+
+            String accessTokenScopeTable = "IDN_OAUTH2_ACCESS_TOKEN_SCOPE";
+            String partitionedAccessTokenScopeTable = OAuth2Util.getPartitionedTableByUserStore(accessTokenScopeTable,
+                    userStoreDomain);
+
+            if (log.isDebugEnabled()) {
+                log.debug("PartitionedAccessTokenTable: " + partitionedAccessTokenTable +
+                        " & PartitionedAccessTokenScopeTable" + partitionedAccessTokenScopeTable +
```

Comment:
Keep a space " & PartitionedAccessTokenScopeTable " +partitionedAccessTokenScopeTable
 for clarity of the log

Reply:
Good catch.. Thanks for pointing that out